src/services/outliningElementsCollector.ts

```diff
@@ -9,7 +9,43 @@ namespace ts.OutliningElementsCollector {
 
     function addNodeOutliningSpans(sourceFile: SourceFile, cancellationToken: CancellationToken, out: Push<OutliningSpan>): void {
         let depthRemaining = 40;
-        sourceFile.forEachChild(function walk(n) {
+        let current = 0;
+        const statements = sourceFile.statements;
+        const n = statements.length;
+        while (current < n) {
+            while (current < n && !isAnyImportSyntax(statements[current])) {
+                visitNonImportNode(statements[current]);
+                current++;
+            }
+            if (current === n) break;
+            const firstImport = statements[current];
+            while (current < n && isAnyImportSyntax(statements[current])) {
+                visitImportNode(statements[current] as AnyImportSyntax, sourceFile, cancellationToken, out);
+                current++;
+            }
+            const lastImport = current < n ? statements[current - 1] : statements[n - 1];

```

Comment:
I think this is always just `current - 1` since as soon as `current === n`... well, yeah